pycbc/events/trigger_fits.py

```diff
@@ -0,0 +1,212 @@
+"""
+Tools for maximum likelihood fits to single trigger statistic values
+
+For some set of values above a threshold, e.g. trigger SNRs, the functions
+in this module perform maximum likelihood fits with 1-sigma uncertainties
+to various simple functional forms of PDF, all normalized to 1.
+You can also obtain the fitted function and its (inverse) CDF and perform
+a Kolmogorov-Smirnov test.
+
+Usage:
+# call the fit function directly if the threshold is known
+alpha, sigma_alpha = fit_exponential(snrs, 5.5)
+
+# apply a threshold explicitly
+alpha, sigma_alpha = fit_above_thresh('exponential', snrs, thresh=6.25)
+
+# let the code work out the threshold from the smallest value via the default thresh=None
+alpha, sigma_alpha = fit_above_thresh('exponential', snrs)
+
+# or only fit the largest N values, i.e. tail fitting
+thresh = tail_threshold(snrs, N=500)
+alpha, sigma_alpha = fit_above_thresh('exponential', snrs, thresh)
+
+# obtain the fitted function directly
+xvals = numpy.xrange(5.5, 10.5, 20)
+exponential_fit = expfit(xvals, alpha, thresh)
+
+# or access function by name
+exponential_fit_1 = fit_fn('exponential', xvals, alpha, thresh)
+
+# get the KS test statistic and p-value - see scipy.stats.kstest
+ks_stat, ks_pval = KS_test('exponential', snrs, alpha, thresh)
+
+"""
+
+# Copyright T. Dent 2015 (thomas.dent@aei.mpg.de)
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 2 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+
+from __future__ import division
+import numpy
+from scipy.stats import kstest
+
+fitalpha_dict = {
+    'exponential' : lambda vals, thresh : 1. / (numpy.mean(vals) - thresh),
+    'rayleigh'    : lambda vals, thresh : 2. / (numpy.mean(vals**2.) - thresh**2.),
+    'power'       : lambda vals, thresh : numpy.mean(numpy.log(vals/thresh))**-1. + 1.
+}
+
+# measurement standard deviation = (-d^2 log L/d alpha^2)^(-1/2)
+fitstd_dict = {
```

Comment:
Tom, are these functions correct? They take a threshold argument but don't actually use it, and they appear to have an undefined alpha value!